healsparse/geom.py

```diff
@@ -233,18 +250,37 @@ def radius(self):
         return self._radius
 
     def get_pixels(self, *, nside):
-        return hpg.query_circle(
-            nside,
+        if self._nside_render is not None:
+            if nside < self._nside_render:
+                raise ValueError(f"Cannot render a Circle with {self._nside_render} into nside={nside}")
+            nside_render = self._nside_render
+        else:
+            nside_render = nside
+
+        pixels = hpg.query_circle(
+            nside_render,
             self._ra,
             self._dec,
             self._radius,
             nest=True,
             inclusive=False,
         )
 
+        if self._nside_render is not None:
```

Comment:
The subtle addition of `self.` here is what short circuts this call when nside_render is not set. Maybe rename the `nside_render` variable above to make things clearer?

Reply:
See my suggested change.